flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriter.java

```diff
@@ -141,15 +141,10 @@ boolean isDone() {
 	void abort(long checkpointId, Throwable cause);
 
 	/**
-	 * Must be called after {@link #start(long, CheckpointOptions)}.
+	 * Must be called after {@link #start(long, CheckpointOptions)} once.
 	 */
 	ChannelStateWriteResult getWriteResult(long checkpointId);
```

Comment:
Please rename `getWriteResult`. I think most developers would expect a getter to be idempotent.

How about `removeWriteResult`? That would be similar to how a Java map works. Or `getAndRemoveWriteResult` to be completely explicit.

Reply:
Good point, I'll rename it to `getAndRemoveWriteResult`.